--- grub-2.05_alpha20200110/include/grub/i386/linux.h.orig	2020-02-14 12:30:43.562177649 -0800
+++ grub-2.05_alpha20200110/include/grub/i386/linux.h	2020-02-14 16:22:05.061348599 -0800
@@ -14,21 +14,21 @@
  *
  *  You should have received a copy of the GNU General Public License
  *  along with GRUB.  If not, see <http://www.gnu.org/licenses/>.
  */
 
 #ifndef GRUB_I386_LINUX_HEADER
 #define GRUB_I386_LINUX_HEADER	1
 
 #define GRUB_LINUX_I386_MAGIC_SIGNATURE	0x53726448      /* "HdrS" */
 #define GRUB_LINUX_DEFAULT_SETUP_SECTS	4
-#define GRUB_LINUX_INITRD_MAX_ADDRESS	0x37FFFFFF
+#define GRUB_LINUX_INITRD_MAX_ADDRESS	0x1ffffffff
 #define GRUB_LINUX_MAX_SETUP_SECTS	64
 #define GRUB_LINUX_BOOT_LOADER_TYPE	0x72
 #define GRUB_LINUX_HEAP_END_OFFSET	(0x9000 - 0x200)
 
 #define GRUB_LINUX_BZIMAGE_ADDR		0x100000
 #define GRUB_LINUX_ZIMAGE_ADDR		0x10000
 #define GRUB_LINUX_OLD_REAL_MODE_ADDR	0x90000
 #define GRUB_LINUX_SETUP_STACK		0x9000
 
 #define GRUB_LINUX_FLAG_BIG_KERNEL	0x1
