--- dict/src/Makefile.in.orig	2014-01-13 13:45:24.197678546 -0800
+++ dict/src/Makefile.in	2014-01-13 13:43:35.844792345 -0800
@@ -398,7 +398,7 @@
 
 stardict_DEPENDENCIES = lib/libstardict.la $(LOCAL_SIGCPP_LIBFILE)
 stardict_LDFLAGS = 
-stardict_LDADD = lib/libstardict.la $(STARDICT_LIBS) $(LOCAL_SIGCPP_LIBFILE)
+stardict_LDADD = lib/libstardict.la $(STARDICT_LIBS) $(LOCAL_SIGCPP_LIBFILE) -lgmodule-2.0
 @GNOME_SUPPORT_FALSE@@MAEMO_SUPPORT_FALSE@CONFIG_FILE_MODULE = 
 @GNOME_SUPPORT_FALSE@@MAEMO_SUPPORT_TRUE@CONFIG_FILE_MODULE = gconf_file.cpp gconf_file.h
 @GNOME_SUPPORT_TRUE@CONFIG_FILE_MODULE = gconf_file.cpp gconf_file.h
