diff -U 3 -dHrN nicolas33-offlineimap-98f5181/offlineimap.py nicolas33-offlineimap-98f5181.new/offlineimap.py
--- nicolas33-offlineimap-98f5181/offlineimap.py	2010-12-09 19:52:48.000000000 +0100
+++ nicolas33-offlineimap-98f5181.new/offlineimap.py	2010-12-23 21:08:32.000000000 +0100
@@ -17,6 +17,7 @@
 #    along with this program; if not, write to the Free Software
 #    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 
+from locale import *
 from offlineimap import OfflineImap
 
 offlineimap = OfflineImap()
