# http://comments.gmane.org/gmane.mail.imap.offlineimap.general/2523

--- offlineimap/imaplib2.py.orig	2012-02-04 12:11:44.000000000 -0800
+++ offlineimap/imaplib2.py	2012-04-26 17:48:54.864753664 -0700
@@ -460,7 +460,7 @@
                 cert_reqs = ssl.CERT_REQUIRED
             else:
                 cert_reqs = ssl.CERT_NONE
-            self.sock = ssl.wrap_socket(self.sock, self.keyfile, self.certfile, ca_certs=self.ca_certs, cert_reqs=cert_reqs)
+            self.sock = ssl.wrap_socket(self.sock, self.keyfile, self.certfile, ca_certs=self.ca_certs, cert_reqs=cert_reqs, ssl_version = ssl.PROTOCOL_TLSv1)
             ssl_exc = ssl.SSLError
         except ImportError:
             # No ssl module, and socket.ssl does not allow certificate verification
