diff --git a/Quake/Makefile b/Quake/Makefile
index 2b09be9..05ca5b7 100644
--- a/Quake/Makefile
+++ b/Quake/Makefile
@@ -5,7 +5,7 @@
 # "make DO_USERDIRS=1" to enable user directories support
 
 # Enable/Disable user directories support
-DO_USERDIRS=0
+DO_USERDIRS=1
 
 ### Enable/Disable SDL2
 USE_SDL2=0
@@ -13,8 +13,8 @@ USE_SDL2=0
 ### Enable/Disable codecs for streaming music support
 USE_CODEC_WAVE=1
 USE_CODEC_FLAC=0
-USE_CODEC_MP3=1
-USE_CODEC_VORBIS=1
+USE_CODEC_MP3=0
+USE_CODEC_VORBIS=0
 USE_CODEC_OPUS=0
 # either mikmod or xmp
 USE_CODEC_MIKMOD=0
@@ -45,7 +45,7 @@ DEBUG   ?= 0
 CC ?= gcc
 LINKER = $(CC)
 
-STRIP ?= strip
+STRIP ?= echo
 
 #CPUFLAGS= -mtune=i686
 #CPUFLAGS= -march=pentium4
