--- text-icu-0.7.0.1/include/hs_text_icu.h	2015-01-08 20:58:03.000000000 -0800
+++ text-icu-0.7.0.1/include/hs_text_icu.h.fixed	2021-08-10 09:45:25.437650560 -0700
@@ -8,20 +8,22 @@
 #include "unicode/uchar.h"
 #include "unicode/ucol.h"
 #include "unicode/ucnv.h"
 #include "unicode/uiter.h"
 #include "unicode/unorm.h"
 #include "unicode/uregex.h"
 #include "unicode/ustring.h"
 
 #include <stdint.h>
 
+#include <stdbool.h>
+
 /* ubrk.h */
 
 UBreakIterator* __hs_ubrk_open(UBreakIteratorType type, const char *locale,
 			       const UChar *text, int32_t textLength,
 			       UErrorCode *status);
 void __hs_ubrk_close(UBreakIterator *bi);
 void __hs_ubrk_setText(UBreakIterator* bi, const UChar *text,
 		       int32_t textLength, UErrorCode *status);
 UBreakIterator * __hs_ubrk_safeClone(const UBreakIterator *bi,
 				     void *stackBuffer, int32_t *pBufferSize,
